src/common/components/elements/CurrentListings.tsx

```diff
@@ -0,0 +1,323 @@
+  const queryFilters: string[] =
+    (router.query['filters[]'] as string[]) || (router.query.filters as string)?.split(',') || [];
+  const querySortBy = (router.query.sort as string) || 'recently_listed';
+
+  const querySearch = router.query.search || '';
+
+  const validFilters = queryFilters
+    .map((qf) => qf.toUpperCase())
+    .filter((qf) => FilterValues.includes(qf as any)) as FilterAction[];
+
+  const validSortBy = (
+    SortByValues.includes(querySortBy.toUpperCase() as any)
+      ? querySortBy.toUpperCase()
+      : 'RECENTLY_LISTED'
+  ) as SortByAction;
+
+  const [state, dispatch] = useReducer(
+    reducer,
+    initialState({
+      filters: validFilters.length ? validFilters : ['SHOW_ALL'],
+      sortBy: validSortBy,
+    })
+  );
+
+  const loadMoreData = () => {
+    if (loading) {
+      return;
+    }
+    setLoading(true);
+    dispatch({ type: 'LOAD_MORE_LISTINGS' });
+    setLoading(false);
+  };
+
+  // get all listings initially
+  useEffect(() => {
+    async function getListings() {
+      const allListings = await callMetaplexIndexerRPC('getListings');
+      dispatch({ type: 'INITIALIZE_LISTINGS', payload: allListings });
+
+      loadMoreData();
+    }
+
+    getListings();
+  }, []);
+
+  const hasNextPage = state.listingsOnDisplay.length < state.filteredListings.length;
+  const [sentryRef] = useInfiniteScroll({
+    loading,
+    hasNextPage,
+    onLoadMore: () => dispatch({ type: 'LOAD_MORE_LISTINGS' }),
+    // When there is an error, we stop infinite loading.
+    // It can be reactivated by setting "error" state as undefined.
+    // disabled: false || !!error, //  !!error,
+
+    // `rootMargin` is passed to `IntersectionObserver`.
+    // We can use it to trigger 'onLoadMore' when the sentry comes near to become
+    // visible, instead of becoming fully visible on the screen.
+    rootMargin: '0px 0px 400px 0px',
+  });
+
+  return (
+    <>
+      <Row justify="space-between" align="middle" style={{ marginBottom: 30 }}>
+        <Title level={3}>
+          Current listings ({state.filteredListings.length}) (
+          {state.filteredListings.length -
+            new Set(state.filteredListings.map((l) => l.address)).size}{' '}
+          duplicates){' '}
+        </Title>
+        {/* <Space direction="horizontal">
+          <DiscoveryFilterDropdown
+            label="Filter"
+            value={state.filters}
+            options={filterOptions}
+            dispatch={dispatch}
+          />
+          <DiscoveryRadioDropdown
+            label="Sort"
+            action="SORT"
+            value={state.sortBy}
+            options={sortingOptions}
+            dispatch={dispatch}
+          />
+        </Space> */}
+        <DiscoveryFiltersAndSortBy
+          sortBy={state.sortBy}
+          filters={state.filters}
+          allFilterOptions={filterOptions}
+          allSortByOptions={sortingOptions}
+          dispatch={dispatch}
+        />
+      </Row>
+
+      <List
```

Comment:
What I'd like to work toward is having the components in the home page and wrapping the filter and infinite scroll into a hook.